frontend/src/components/pages/RegistrantExperience/RegistrationSteps/ReviewRegistration/EditCamperCard.tsx

```diff
@@ -0,0 +1,409 @@
+import React, { useMemo, useState } from "react";
+import {
+  Box,
+  Button,
+  Divider,
+  Flex,
+  FormControl,
+  FormLabel,
+  Heading,
+  Input,
+  Spacer,
+  Text,
+  Textarea,
+  useDisclosure,
+  useToast,
+  Wrap,
+  WrapItem,
+  FormErrorMessage,
+} from "@chakra-ui/react";
+import {
+  PersonalInfoActions,
+  PersonalInfoReducerDispatch,
+} from "../../../../../types/PersonalInfoTypes";
+import RequiredAsterisk from "../../../../common/RequiredAsterisk";
+import { RegistrantExperienceCamper } from "../../../../../types/CamperTypes";
+import {
+  checkAge,
+  checkFirstName,
+  checkLastName,
+} from "../PersonalInfo/personalInfoReducer";
+import DeleteModal from "../../../../common/DeleteModal";
+import { CampResponse } from "../../../../../types/CampsTypes";
+
+type EditCamperCardProps = {
+  camper: RegistrantExperienceCamper;
+  camperIndex: number;
+  camp: CampResponse;
+  dispatchPersonalInfoAction: (action: PersonalInfoReducerDispatch) => void;
+};
+
+const EditCamperCard = ({
+  camper,
+  camperIndex,
+  dispatchPersonalInfoAction,
+  camp,
+}: EditCamperCardProps): React.ReactElement => {
+  const [updateMemo, setUpdateMemo] = useState(0);
+  const initialCamper: RegistrantExperienceCamper = useMemo(() => camper, [
```

Comment:
could you clarify why we need useMemo here? We are not really calculating or processing anything here are we?

Reply:
So basically in the `EditCamperCard` and `EditContactCard` components, we pass in the current state of the `camper` object as a prop. We might edit the campers and contacts using these edit cards, but if we click cancel, we need a way to keep track of the initial camper and contacts and set our current edit campers/contacts to those old ones. This isn't as straightforward as just making a copy/deep copy since any edits to the camper object will in turn rerender this edit component and rerender that copy too, updating it (which we don't want).

I use `useMemo` to keep track of the initial camper/contact object so that if a registrant edits it, the initial object isn't changed, and if someone cancels, the object can be set to the initial object. In the dependency array, I have a state `updateMemo` which only updates when a registrant clicks Save, which is when we want a new initial object. Hope that made sense

Reply:
thanks for explaining!